effectful/handlers/pyro.py

```diff
@@ -534,8 +534,8 @@ def _embed_masked(d) -> Term[TorchDistribution]:
 @defterm.register(dist.LogisticNormal)
 @defterm.register(dist.Normal)
 @defterm.register(dist.StudentT)
-def _embed_loc_scale(d) -> Term[TorchDistribution]:
-    return _DistributionTerm(type(d), d.loc, d.scale)
+def _embed_loc_scale(d: TorchDistribution) -> Term[TorchDistribution]:
+    return defdata(_register_distribution_op(type(d)), d.loc, d.scale)
```

Comment:
Once the relevant `Term` constructor for a semantic type has been registered with `defdata`, you shouldn't even need to invoke `defdata` explicitly - all these calls to `defdata` in all of the `defterm` patterns here can be replaced with applications of the operation, with `defdata` invoked implicitly via `Operation.__free_rule__`:
```diff
- return defdata(_register_distribution_op(type(d)), d.loc, d.scale)
+ return _register_distribution_op(type(d))(d.loc, d.scale)
```